kube-client/src/config/file_config.rs

```diff
@@ -72,20 +73,24 @@ pub struct NamedExtension {
 }
 
 /// NamedCluster associates name with cluster.
-#[derive(Clone, Debug, Serialize, Deserialize)]
+#[serde_as]
+#[derive(Clone, Debug, Serialize, Deserialize, Default)]
 #[cfg_attr(test, derive(PartialEq, Eq))]
 pub struct NamedCluster {
     /// Name of cluster
+    #[serde(default)]
     pub name: String,
     /// Information about how to communicate with a kubernetes cluster
+    #[serde_as(as = "serde_with::DefaultOnNull")]
     pub cluster: Cluster,
```

Comment:
I think using an `Option` is a lot clearer and avoids a dependency 👍

Reply:
I opted for the DefaultOnNull to avoid breaking changes. Also from a consumer perspective, I find working with non-null structs much simpler and overall better experience, especially these ones that are rarely null.

That's just my opinion, but I'm easy, if majority prefers to use Option and remove the dependency, I can change it 👍

Reply:
I am a bit torn on it myself. It's not a dependency that is in our recursive dependency tree so it will be a new dependency for many. If it's ultimately just a slight inconvenience to have to `.unwrap_or_default` during construction phase for us internally (and for the small amount of people doing manual work on our structs), then it is possibly worth keeping defaulting and parsing separate (i.e. using options + unwrap_or_default).

Anyone else have opinions? Maybe @kazk ?

Reply:
I think we should use `Option<T>`.

- Defaulting on null/missing is a surprising behavior in Rust, at least to me. It's more idiomatic to use `Option<T>`.
- `Option<T>` is easier to check if the value is present when we need them. We can also return a better error when missing. If we default on missing, we can't tell if it was missing in the config or default was provided.
- Changing to `Option<T>` is a breaking change, but it's a small and straightforward change. It should also be mostly internal.

Reply:
Thanks all for the feedback, I changed to Option<T> now and removed serde_with.